test/nodes/test_shaper.py

```diff
@@ -0,0 +1,563 @@
+    assert isinstance(questions, list) and len(questions) == 2 and questions[0] == result["query"]
+
+
+def test_basic_function_invocation_no_output_var(tmp_path):
+    with open(tmp_path / "tmp_config.yml", "w") as tmp_file:
+        tmp_file.write(
+            f"""
+            version: ignore
+            components:
+            - name: shaper
+              params:
+                inputs:
+                    query:
+                      func: expand
+                      output: questions
+                      params:
+                        expand_target: query
+                        size:
+                          func: len
+                          params:
+                            - documents
+                    documents:
+                      func: concat_docs
+                      #output: if output is not specified the result will be bound to input variable (documents)
+                      params:
+                        docs: documents
+                        delimiter: " "
+              type: Shaper
+            pipelines:
+              - name: query
+                nodes:
+                  - name: shaper
+                    inputs:
+                      - Query
+        """
+        )
+    pipeline = Pipeline.load_from_yaml(path=tmp_path / "tmp_config.yml")
+    result = pipeline.run(
+        query="What can you tell me about Berlin?",
+        documents=[Document("Berlin is an amazing city."), Document("I love Berlin.")],
+    )
+    assert result
+    assert isinstance(result["query"], str)
+    questions = result["meta"]["invocation_context"]["questions"]
+
+    # questions has been expanded to a list of strings of size 2 (because Documents has 2 elements)
+    assert isinstance(questions, list) and len(questions) == 2 and questions[0] == result["query"]
+    docs = result["documents"]
+    assert isinstance(docs, str) and docs == "Berlin is an amazing city. I love Berlin"
+
+
+def test_rename_vars(tmp_path):
+    with open(tmp_path / "tmp_config.yml", "w") as tmp_file:
+        tmp_file.write(
+            f"""
+            version: ignore
+            components:
+            - name: shaper
+              params:
+                inputs:
+                    query:
+                      output: questions
+              type: Shaper
+            pipelines:
+              - name: query
+                nodes:
+                  - name: shaper
+                    inputs:
+                      - Query
+        """
+        )
+    pipeline = Pipeline.load_from_yaml(path=tmp_path / "tmp_config.yml")
+    result = pipeline.run(
+        query="What can you tell me about Berlin?",
+        documents=[Document("Berlin is an amazing city."), Document("I love Berlin.")],
+    )
+    assert result
+    # query has been renamed to questions
+    assert isinstance(result["meta"]["invocation_context"]["questions"], str)
```

Comment:
Could you add an assert here as well checking that `questions` is the same as `query` to make sure it hasn't been changed?

Reply:
Done